### DISCUSSION TITLE
move from intel to m1 with timemachine, homebrew still work, will this be a problem?

### DISCUSSION THREAD

**Discussion by VRgWfapFT** [Homebrew/discussions#3331] [Casks] [answered]
hi,
I bought a **new m1 computer**, and since there was a lot of data and environment on my intel laptop that needed to be migrated, I directly used Apple's time machine to restore it, **which means i migrate from intel to m1 with time machine**

After the migration, homebrew is still in the `/usr/local/bin` directory, and commands such as `brew update`/`brew install` still work well. The following is the output of `brew config`

HOMEBREW_VERSION: 3.4.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: 30e3d8883cde265cbc16ff94b49a3ac14137590e
Last commit: 4 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 16aca6a7641d00d1121cad88670a1d13d8edb13f
Core tap last commit: 20 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 20
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: 20-core 64-bit westmere
Clang: 13.1.6 build 1316
Git: 2.32.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.4-x86_64
CLT: 13.4.0.0.1.1651278267
Xcode: N/A
Rosetta 2: true

so in this case, 
1. do i need to reinstall homebrew to change the path from `/usr/local/` to the `/opt` directory?  
2. the current program is running well, can the status quo be maintained? the whole environment is involved too much

**Comment by SMillerDev** [comment, marked as answer]
You will need to reinstall homebrew if you want the software to run natively. Anything in `/usr/local` will run through Rosetta. There might also be other issues with time machine backups of homebrew but I don't have a list of those.

The suggested way would be `brew bundle dump` and then `brew bundle install` with the Brewfile on the new device.

**Reply by VRgWfapFT** [reply]
thanks

**Comment by Liopun** [comment]
If anyone is currently having this problem in 2023, you can use this tool https://github.com/Liopun/prp
